packages/react-next/src/components/TeachingBubble/TeachingBubbleContent.base.tsx

```diff
@@ -86,6 +64,26 @@ export class TeachingBubbleContentBase extends React.Component<ITeachingBubblePr
       secondaryButtonClassName: secondaryButtonProps ? secondaryButtonProps.className : undefined,
     });
 
+    const onKeyDown = React.useCallback(
+      (ev: React.KeyboardEvent<HTMLElement> | KeyboardEvent): void => {
+        if (props.onDismiss) {
+          // eslint-disable-next-line deprecation/deprecation
+          if (ev.which === KeyCodes.escape) {
+            props.onDismiss(ev);
+          }
+        }
+      },
+      [props.onDismiss],
```

Comment:
You should reference `onDismiss` directly rather than from props. That way the callback won't be re-created as often. (The identity of the `props` object will change every time props are updated, even if the identity of `onDismiss` stays constant.)
```suggestion
    const onKeyDown = React.useCallback(
      (ev: React.KeyboardEvent<HTMLElement> | KeyboardEvent): void => {
        if (onDismiss) {
          // eslint-disable-next-line deprecation/deprecation
          if (ev.which === KeyCodes.escape) {
           onDismiss(ev);
          }
        }
      },
      [onDismiss],
```